### PR TITLE
search job and company

### PR DESCRIPTION
**Pull request by quanglx-noovado**
## Related Tickets:
- [#32114](https://edu-redmine.sun-asterisk.vn/issues/32114)

## WHAT this PR does?
Search Job and Company

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by quanglx-noovado** [comment]
ready

File: resources/js/search.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+$("document").ready(function () {
+    $("input[name=tag]").on("click", countChecked);
+    var countChecked = function () {
+        var checkedValue = $('.checkbox:checked').val();
+        console.log(checkedValue)
```
Comment (congnx-2423) [inline resources/js/search.js]: bỏ console.log nha e

File: resources/views/layouts/list_job.blade.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                        <p>{{ $job->company->address }}</p>
+                        <span class="meta"></span>
+                    </div>
+                    <div class="clearfix"> </div>
+                    <br>
+                    <div class="col-md-6 single_right">
+                        <p>
+                            <b>@lang('job.company'): </b>
+                            {{ $job->company->name }}
+                        </p>
```
Comment (congnx-2423) [inline resources/views/layouts/list_job.blade.php]: bên lấy data lấy cả with('tag') tránh n+1 e nha

File: resources/views/search_company.blade.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                                            <br>
+                                            <div class="col-md-6 single_right">
+                                                <p>
+                                                    <b>@lang('job.contact'): </b>
+                                                    {{ $company->user->name }}
+                                                </p>
+                                            </div>
+                                            <div class="col-md-6 single_right">
+                                                <p>
+                                                    <b>@lang('job.email'): </b>
```
Comment (congnx-2423) [inline resources/views/search_company.blade.php:37]: dấu ':' dùng &#58;

File: routes/web.php [hunk 4]
```diff
@@ -36,3 +36,6 @@
 Route::get('list-candidate/{id}','JobController@showListCandidateApply')->name('list_candidate');
 
 Route::get('accept-reject/{user_id}/{job_id}/{status}','JobController@acceptOrReject')->name('accept_reject');
+
+Route::get('/search','JobController@search')->name('search');
+Route::post('filter','JobController@filter')->name('filter');
```
Comment (congnx-2423) [inline routes/web.php]: xuống 1 dòng nha e

**Review by huyenttm-0901 — COMMENTED** [review]
conflicted

**Comment by quanglx-noovado** [comment]
ready

**Review by huyenttm-0901 — COMMENTED** [review]
conflicts

**Comment by quanglx-noovado** [comment]
ready

File: app/Http/Controllers/JobController.php [hunk 5]
```diff
@@ -143,11 +145,35 @@ public function acceptOrReject($userId, $jobId, $status)
...
     }
+
+    public function search(Request $request)
+    {
+        if ($request->title) {
+            $jobs = Job::where('title', 'LIKE', '%' . $request->title . '%')->get();
+            foreach ($jobs as $job) {
+                $job->url =  $job->company->images()->where('type', config('user.avatar'))->first();
+            }
+
```
Comment (huyenttm-0901) [inline app/Http/Controllers/JobController.php]: N+1 query

File: app/Http/Controllers/JobController.php [hunk 6]
```diff
@@ -143,11 +145,35 @@ public function acceptOrReject($userId, $jobId, $status)
...
+            return view('search_jobs', [
+                'jobs' => $jobs,
+            ]);
+        }
+
+        $companies = Company::where('name', 'LIKE', '%' . $request->name . '%')->get();
+
+        foreach ($companies as $company) {
+            $company->url =  $company->images()->where('type', config('user.avatar'))->first();
+        }
```
Comment (huyenttm-0901) [inline app/Http/Controllers/JobController.php]: tương tự

File: resources/views/layouts/search.blade.php [hunk 7]
```diff
@@ -4,9 +4,11 @@
             <div id="search_form" class="clearfix">
                 <h1>@lang('home.search_title')</h1>
                 <p>
-                    <input type="text" class="text" placeholder="@lang('home.search_keyword')" value="">
-                    <input type="text" class="text" placeholder="@lang('home.search_company')" value="">
-                    <label class="btn2 btn-2 btn2-1b"><input type="submit" value="@lang('home.search_find')"></label>
+                <form action="{{route('search')}}" method="GET">
```
Comment (huyenttm-0901) [inline resources/views/layouts/search.blade.php]: convention

File: resources/views/search_company.blade.php [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    <div class="container">
+        <div class="single">
+            <div class="col-md-10 single_right">
+                <div class="but_list">
+                    <div class="bs-example bs-example-tabs" role="tabpanel" data-example-id="togglable-tabs">
+                        <div id="myTabContent" class="tab-content">
+                            <div role="tabpanel" class="tab-pane fade in active" id="home" aria-labelledby="home-tab">
+                                @foreach ($companies as $company)
+                                    <div class="jobs-item with-thumb">
+                                    <div class="thumb"><a href="{{asset($company->url)}}"><img src="" class="img-responsive" alt="" /></a></div>
```
Comment (huyenttm-0901) [inline resources/views/search_company.blade.php]:
```suggestion
                                    <div class="thumb"><a href="{{ asset($company->url) }}"><img src="" class="img-responsive" alt="" /></a></div>
```

**Comment by quanglx-noovado** [comment]
ready

**Pull request merged**